authority/provisioner/sign_ssh_options.go

```diff
@@ -46,13 +46,22 @@ type SSHCertificateOptionsValidator interface {
 	Valid(got SSHOptions) error
 }
 
+// sshModifierFunc is an adapter to allow the use of ordinary functions as SSH
+// certificate modifiers.
+type sshModifierFunc func(cert *ssh.Certificate) error
+
+func (f sshModifierFunc) Modify(cert *ssh.Certificate) error {
```

Comment:
Is this just for readability or is there a more formal reason why adding an alternate way to invoke `f` using its type alias is beneficial?

Reply:
All the currently supported modifiers/validators are methods on types. With the backdate I needed to change a couple of methods that used to be `SSHCertificateModifier` to be a `SSHCertificateOptionModifier` so I can pass some extra options:

```go
type SSHCertificateModifier interface {
	SignOption
	Modify(cert *ssh.Certificate) error
}

type SSHCertificateOptionModifier interface {
	SignOption
	Option(o SSHOptions) SSHCertificateModifier
}
```

Adding that type I can easily convert just a function to a value with the `SSHCertificateOptionModifier` type, and I don't really need to add two extra types to implement those methods.

This is also common in Go, a great example is `http.HandlerFunc`, see:
https://golang.org/pkg/net/http/#HandlerFunc

Reply:
I think I get it but I'm sure it will become more clear as I see more go code.